components/post_view/post_view.jsx

```diff
@@ -299,7 +293,7 @@ export default class PostView extends React.PureComponent {
         //This can be different than the unreadCount on the sidebar as sytem messages
         //are not considered for the count.
         return posts.reduce((count, post) => {
-            if (post.create_at > this.state.lastViewedAt &&
+            if (post.create_at > this.props.lastViewedAt &&
```

Comment:
Totally unrelated now, but I just realized that using `this.state.lastViewedAt` in the old code might be what causes the bug where switching from an unread channel to a read one and then hitting back on the browser didn't clear the New Messages line